### PR TITLE
Feature/iep 504 validate companyname with alphakey

### PR DESCRIPTION
**Pull request by gshivach**
company name in title case were rejected by the api. added check to compare name against company profile api using the alpha key service - sameAsAlphaKey.
amended code to validate all other company details only if the company existed in company profile with a name and number match.

### CODE REVIEW

File: handlers/insolvency_resource.go [hunk 1]
```diff
@@ -69,15 +70,33 @@ func HandleCreateInsolvencyResource(svc dao.Service) http.Handler {
...
+		var companyProfile *companieshouseapi.CompanyProfile
+		err, httpStatus, companyProfile = service.CheckCompanyExists(&request, req)
 		if err != nil {
 			log.ErrorR(req, fmt.Errorf("company was not found valid when checking company profile API [%v]", err))
 			m := models.NewMessageResponse(fmt.Sprintf("company [%s] was not found valid for insolvency: %v", request.CompanyNumber, err))
 			utils.WriteJSONWithStatus(w, req, m, httpStatus)
 			return
 		}
 
+		//Check with alphakey service if company name valid and Check with company profile API if other details are valid
```
Comment (jphillips2121) [inline handlers/insolvency_resource.go:83]: Needs a space between `//` and `Check` like other comments

File: handlers/insolvency_resource.go [hunk 2]
```diff
@@ -69,15 +70,33 @@ func HandleCreateInsolvencyResource(svc dao.Service) http.Handler {
...
+		err, httpStatus = service.CheckCompanyNameAlphaKey(companyProfile.CompanyName, &request, req)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("company name was not found valid when checking company profile API [%v]", err))
+			m := models.NewMessageResponse(fmt.Sprintf("company name [%s] was not found valid for insolvency: %v", request.CompanyNumber, err))
+			utils.WriteJSONWithStatus(w, req, m, httpStatus)
+			return
+		} else {
+			err = service.CheckCompanyDetailsAreValid(companyProfile, &request)
```
Comment (jphillips2121) [inline handlers/insolvency_resource.go:91]: This check doesn't need to be within an `else` statement. The `else` statement can be removed

File: service/alpha_key_service.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		return fmt.Errorf("error verifying company name"), requestAlphaKeyResponse.HTTPStatusCode
+	}
+
+	profileAlphaKey := profileAlphaKeyResponse.SameAsAlphaKey
+
+	if !strings.EqualFold(insolvencyRequestAlphaKey, profileAlphaKey) {
+		return fmt.Errorf("company names do not match - provided: [%s], expected: [%s]", insolvencyRequest.CompanyName, companyProfileCompanyName), http.StatusBadRequest
+	}
+
+	//no error
```
Comment (jphillips2121) [inline service/alpha_key_service.go:40]: Don't need a comment here

File: service/company_service.go [hunk 4]
```diff
@@ -64,12 +58,7 @@ func GetCompanyIncorporatedOn(companyNumber string, req *http.Request) (string,
 }
 
 // checkCompanyDetailsAreValid checks the incoming company profile to see if it's valid for insolvency
-func checkCompanyDetailsAreValid(companyProfile *companieshouseapi.CompanyProfile, insolvencyRequest *models.InsolvencyRequest) error {
-
-	// Check company name in request and company profile match
-	if !strings.EqualFold(companyProfile.CompanyName, insolvencyRequest.CompanyName) {
-		return fmt.Errorf("company names do not match - provided: [%s], expected: [%s]", insolvencyRequest.CompanyName, companyProfile.CompanyName)
-	}
+func CheckCompanyDetailsAreValid(companyProfile *companieshouseapi.CompanyProfile, insolvencyRequest *models.InsolvencyRequest) error {
```
[the remark this answers is not in the document]
Comment (gshivach) [reply service/company_service.go:61]: Done

[the remark this answers is not in the document]
Comment (gshivach) [reply service/company_service.go:61]: done

File: service/alpha_key_service.go (hunk 3 above)
Comment (gshivach) [reply service/alpha_key_service.go:40]: done

File: service/alpha_key_service.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		log.ErrorR(req, fmt.Errorf("error communicating with alphakey service [%v]", err))
+		return fmt.Errorf("error verifying company name"), requestAlphaKeyResponse.HTTPStatusCode
+	}
+
+	insolvencyRequestAlphaKey := requestAlphaKeyResponse.SameAsAlphaKey
+
+	profileAlphaKeyResponse, err := api.AlphaKey.Get(companyProfileCompanyName).Do()
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("error communicating with alphakey service [%v]", err))
+		return fmt.Errorf("error verifying company name"), requestAlphaKeyResponse.HTTPStatusCode
```
[the remark this answers is not in the document]
Comment (gshivach) [reply service/alpha_key_service.go:31]: done

File: service/alpha_key_service.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+	api, err := manager.GetPrivateSDK(req)
+	if err != nil {
+		return fmt.Errorf("error creating private SDK to call alphakeyservice: [%v]", err.Error()), http.StatusInternalServerError
+	}
+
+	requestAlphaKeyResponse, err := api.AlphaKey.Get(insolvencyRequest.CompanyName).Do()
+	if err != nil {
+		log.ErrorR(req, fmt.Errorf("error communicating with alphakey service [%v]", err))
+		return fmt.Errorf("error verifying company name"), requestAlphaKeyResponse.HTTPStatusCode
```
[the remark this answers is not in the document]
Comment (gshivach) [reply service/alpha_key_service.go:23]: done

File: handlers/insolvency_resource.go (hunk 1 above)
Comment (gshivach) [reply handlers/insolvency_resource.go:83]: done

File: handlers/insolvency_resource.go (hunk 2 above)
Comment (gshivach) [reply handlers/insolvency_resource.go]: amended

Comment (gshivach) [reply handlers/insolvency_resource.go]: amended

File: service/alpha_key_service.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	insolvencyRequestAlphaKey := requestAlphaKeyResponse.SameAsAlphaKey
+
+	profileAlphaKeyResponse, err := api.AlphaKey.Get(companyProfileCompanyName).Do()
```
[the remark this answers is not in the document]
Comment (gshivach) [reply service/alpha_key_service.go:28]: added test

**Review by emead — APPROVED** [review]

**Pull request merged**